tests/operators/test_agnostic_append.py

```diff
@@ -0,0 +1,116 @@
+import pathlib
+
+import pandas as pd
+from airflow.exceptions import BackfillUnfinished
+
+from astro import sql as aql
+from astro.dataframe import dataframe as adf
+from tests.operators import utils as test_utils
+
+CWD = pathlib.Path(__file__).parent
+import pytest
+
+
+@adf
+def validate_basic(df: pd.DataFrame):
+    assert len(df) == 6
+    assert not df["sell"].hasnans
+    assert df["rooms"].hasnans
+
+
+@adf
+def validate_append_all(df: pd.DataFrame):
+    assert len(df) == 6
+    assert not df["sell"].hasnans
+    assert not df["rooms"].hasnans
+
+
+@adf
+def validate_caste_only(df: pd.DataFrame):
+    assert len(df) == 6
+    assert not df["age"].hasnans
+    assert df["sell"].hasnans
+
+
+@pytest.fixture
+def append_params(request):
+    mode = request.param
+    if mode == "basic":
+        return {
+            "columns": ["sell", "living"],
+        }, validate_basic
+    if mode == "all_fields":
+        return {}, validate_append_all
+    if mode == "with_caste":
+        return {
+            "columns": ["sell", "living"],
+            "casted_columns": {"age": "INTEGER"},
+        }, validate_basic
+    if mode == "caste_only":
+        return {"casted_columns": {"age": "INTEGER"}}, validate_caste_only
+
+
+@pytest.mark.parametrize(
+    "append_params",
+    ["basic", "all_fields", "with_caste"],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "sql_server",
+    [
+        "bigquery",
+        "snowflake",
+        "postgres",
+        "sqlite",
+    ],
+    indirect=True,
+)
+def test_append(sql_server, sample_dag, tmp_table, append_params):
+    app_param, validate_append = append_params
+
+    with sample_dag:
+        load_main = aql.load_file(
```

Comment:
can we move loading files into tmp_table? is there any usecase with just tmp_table without loading it to db?

Reply:
For now using load_file ensures that the actual function is being passed info from a previous task instead of a table object